### PR TITLE
Add io.github.Bollos00.LibreMines

### PR DESCRIPTION
**Pull request by Bollos00**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.github.Bollos00.LibreMines.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+    "app-id": "io.github.Bollos00.LibreMines",
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.15",
```
Comment (hfiguiere) [inline io.github.Bollos00.LibreMines.json:4]:
```suggestion
    "runtime-version": "5.15-21.08",
```

File: io.github.Bollos00.LibreMines.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "sdk": "org.kde.Sdk",
+    "command": "libremines",
+    "rename-desktop-file": "libremines.desktop",
+    "rename-icon": "libremines",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=fallback-x11",
+        "--socket=wayland",
```
Comment (hfiguiere) [inline io.github.Bollos00.LibreMines.json:13]:
```suggestion
        "--socket=fallback-x11",
        "--socket=wayland",
```

File: io.github.Bollos00.LibreMines.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        "--filesystem=host",
```
Comment (hfiguiere) [inline io.github.Bollos00.LibreMines.json:14]: Why do you need host filesystem access?

**Comment by hfiguiere** [comment]
bot, build io.github.Bollos00.LibreMines

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
build failed because it can't find the appdata file.

File: io.github.Bollos00.LibreMines.json (hunk 3 above)
Comment (Bollos00) [reply io.github.Bollos00.LibreMines.json:14]: I followed the [example of the Qt application on the documentation](https://docs.flatpak.org/en/latest/qt.html).

I can test building the package without that line if you prefer.

**Comment by Bollos00** [comment]
> build failed because it can't find the appdata file.

I missed the appdata, I will provide it and let you know.

**Comment by Bollos00** [comment]
I made the appdata file and place it on the repository. It passed on the test (appstream-util validate-relax io.github.Bollos00.LibreMines.appdata.xml) so I think is okay.

File: io.github.Bollos00.LibreMines.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<component type="desktop">
+  <id>io.github.Bollos00.LibreMines</id>
+  <launchable type="desktop-id">libremines</launchable>
+  <metadata_license>CC-BY-SA-4.0</metadata_license>
```
Comment (hfiguiere) [inline io.github.Bollos00.LibreMines.appdata.xml:5]: we prefer `CC0-1.0`

I don't even think CC-BY-SA will pass the validator

File: io.github.Bollos00.LibreMines.json (hunk 3 above)
Comment (hfiguiere) [reply io.github.Bollos00.LibreMines.json:13]: See Item 3 in the check list you "checked"

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build io.github.Bollos00.LibreMines

**Comment by hfiguiere** [comment]
And he appadata file should go upstream for next release.

**Comment by hfiguiere** [comment]
> I made the appdata file and place it on the repository. It passed on the test (appstream-util validate-relax io.github.Bollos00.LibreMines.appdata.xml) so I think is okay.

You have to install it too.

**Comment by Bollos00** [comment]
> > I made the appdata file and place it on the repository. It passed on the test (appstream-util validate-relax io.github.Bollos00.LibreMines.appdata.xml) so I think is okay.
> 
> You have to install it too.

I see it now. I updated the manifest and it seems to be working now.

**Comment by Bollos00** [comment]
bot, build io.github.Bollos00.LibreMines

File: io.github.Bollos00.LibreMines.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (barthalion) [inline io.github.Bollos00.LibreMines.appdata.xml:1]: Ideally, you should maintain it in your GitHub repo.

File: io.github.Bollos00.LibreMines.json (hunk 3 above)
Comment (barthalion) [reply io.github.Bollos00.LibreMines.json:13]: At first glance, nothing in the app seems to require the host access. Please remove it.

**Comment by Bollos00** [comment]
bot, build io.github.Bollos00.LibreMines

**Comment by hfiguiere** [comment]
same question, why do you need access to the home directory? 

The configuration directory is handled automatically, and at worse, if you don't save the config in an XDG directory you can use `--persist`. I can't imagine why a mime sweeper game would need access to the whole home directory.

**Comment by Bollos00** [comment]
I was about to comment about it.

The game saves the scores, game preferences and idioms on `~/.local/share/libremines` directory. When I just removed the `"--filesystem=host",` line, it gave a file error permission error when trying to load the settings and game scores.

```
QIODevice::read (QFile, "/home/bollos00/.local/share/libreminesDefaultLanguage.txt"): device not open
QIODevice::read (QFile, "/home/bollos00/.local/share/libreminesDefaultLanguage.txt"): device not open
QIODevice::read (QFile, "/home/bollos00/.local/share/libreminesDefaultLanguage.txt"): device not open
QIODevice::read (QFile, "/home/bollos00/.local/share/libreminesDefaultLanguage.txt"): device not open

```

**Comment by hfiguiere** [comment]
then was you want is `--filesystem=xdg-data/libremines` 

But the app should resolve the xdg-directory path instead of hardcoding them, using `QStandardPaths::AppLocalDataLocation` ; and it wouldn't be necessary to add `--filesystem`.

**Comment by hfiguiere** [comment]
> then was you want is `--filesystem=xdg-data/libremines`

Maybe `--filesystem=xdg-data/libremines:create` to make sure it exists.

**Comment by Bollos00** [comment]
> then was you want is `--filesystem=xdg-data/libremines`
> 
> But the app should resolve the xdg-directory path instead of hardcoding them, using `QStandardPaths::AppLocalDataLocation` ; and it wouldn't be necessary to add `--filesystem`.

This is a better way of doing it by the way, I will implement it with `QStandardPaths` later. Thanks.

I will try to build the package with `--filesystem=xdg-data/libremines:create`.

**Comment by Bollos00** [comment]
Well, it does not create the directory during the installation, but the app works as it should even if the directory did not exist in the first place.

**Comment by nedrichards** [comment]
bot, build io.github.Bollos00.LibreMines

**Comment by barthalion** [comment]
Any chance you can land the change to use `QStandardPaths` before we merge it?

**Comment by Bollos00** [comment]
> Any chance you can land the change to use `QStandardPaths` before we merge it?

Sure, I will implement it and make a new release until the weekend.

**Comment by Bollos00** [comment]
bot, build io.github.Bollos00.LibreMines

**Comment by Bollos00** [comment]
I think it is good now.

**Comment by Bollos00** [comment]
Great. Thank you for the review guys.

**Pull request closed without merging**